stacker/actions/diff.py

```diff
@@ -0,0 +1,181 @@
+import logging
+
+from .. import exceptions
+from ..plan import COMPLETE, SKIPPED, Plan
+from . import build
+import difflib
+import json
+
+logger = logging.getLogger(__name__)
+
+
+def diff_dictionaries(old_dict, new_dict):
+    """Diffs two single dimension dictionaries
+
+    Returns the number of changes and an unordered list
+    expressing the common entries and changes.
+
+    Args:
+        old_dict(dict): old dictionary
+        new_dict(dict): new dictionary
+
+    Returns: list()
+        int: number of changed records
+        list: [str(<change type>), <key>, <value>]
+
+        Where <change type>: +, - or empty string
+    """
+
+    old_set = set(old_dict)
+    new_set = set(new_dict)
+
+    added_set = new_set - old_set
+    removed_set = old_set - new_set
+    common_set = old_set & new_set
+
+    changes = 0
```

Comment:
I don't return just the changed lines, but rather all the lines since I felt that helped with context.  For example, I have some parameters passed into my stack which are related and updating one may require updating the other.  This allows me to detect not updating them together when necessary.